Set directv unavailable state when errors returned for longer then a minute

ehendrix23 opened the pull request:
## Description:

Change setting to unavailable when getting request exceptions only after 1 minute has past since 1st occurrence, preventing state set to unavailable when powering on/off.

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [X] There is no commented out code in this PR.

If the code does not interact with devices:
  - [X] Tests have been added to verify that the new code works.

ehendrix23 committed:
Fix unavailable

Change setting to unavailable when getting request exceptions only after 1 minute has past since 1st occurrence.

MartinHjelmare commented on homeassistant/components/media_player/directv.py:
```diff
@@ -165,13 +167,38 @@ def update(self):
-                    self._available = False
+                    # If an error is received then only set to unavailable if
+                    # this started at least 1 minute ago.
+                    if not self._first_error_timestamp:
```
Can we make a small helper method for the common parts of the if else block so we don't need to repeat everything twice?

ehendrix23 replied:
Absolutely. :-) Done.

ehendrix23 committed:
Put common code in _check_state_available

Put common code to determine if available should be set to False in method _check_state_available
